File: Practica6.py

from Circulo import Circulo
from Rectangulo import Rectangulo
from Triangulo import Triangulo
from Hexagono import Hexagono

print("Bienvenido a la calculadora para figuras")

while True:
    print("Elija de qué figura desea calcular el área y el perímetro")
    print("1. Triángulo")
    print("2. Círculo")
    print("3. Rectángulo")
    print("4. Hexágono")
    print("0. Salir")

    opcion = int(input("Ingrese la opción que desea: "))

    if opcion == 1:
        base = float(input("Teclee la base del triángulo: "))
        altura = float(input("Teclee la altura del triángulo: "))
        lado1 = float(input("Teclee el primer lado del triángulo: "))
        lado2 = float(input("Ingrese el segundo lado del triángulo: "))
        lado3 = float(input("Ingrese el tercer lado del triángulo: "))
        triangulo = Triangulo(base, altura, lado1, lado2, lado3)
        print(triangulo)
        print("El área del triángulo es:", triangulo.area())
        print("El perímetro del triángulo es:", triangulo.perimetro())
        print("")
        print("")
        print("")
        print("")

    elif opcion == 2:
        radio = float(input("Teclee el radio del círculo: "))
        circulo = Circulo(radio)
        print(circulo)
        print("El área del círculo es:", circulo.area())
        print("El perímetro del círculo es:", circulo.perimetro())
        print("")
        print("")
        print("")
        print("")

    elif opcion == 3:
        base = float(input("Teclee la base del rectángulo: "))
        altura = float(input("Teclee la altura del rectángulo: "))
        rectangulo = Rectangulo(base, altura)
        print(rectangulo)
        print("El área del rectángulo es:", rectangulo.area())
        print("El perímetro del rectángulo es:", rectangulo.perimetro())
        print("")
        print("")
        print("")
        print("")

    elif opcion == 4:
        lado = float(input("Teclee el valor de un lado del hexágono: "))
        apotema = float(input("Teclee el valor del apotema: "))
        hexagono = Hexagono(lado, apotema)
        print(hexagono)
        print("El área del hexágono es:", hexagono.area())
        print("El perímetro del hexágono es:", hexagono.perimetro())
        print("")
        print("")
        print("")
        print("")

    elif opcion == 0:
        print("Gracias por usar la calculadora de figuras")
        break

    else:
        print("Opción inválida")
        print("")
        print("")
        print("")
        print("")  

File: Practica6.java
import java.util.Scanner;

public class Practica6 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int opc;
        
        do{
            System.out.println("Bienvenido a la calculadora para figuras");
            System.out.println("Elija de que figura desea calcular el area y el perimetro");
            System.out.println("1. Triangulo");
            System.out.println("2. Circulo");
            System.out.println("3. Rectangulo");
            System.out.println("4. Hexagono");
            System.out.println("0. Salir");
            
            opc = scanner.nextInt();
            
            switch(opc){
                case 1 -> {
                    System.out.println("Teclee la base del triangulo");
                    double base = scanner.nextDouble();
                    System.out.println("Teclee la altura del triangulo");
                    double altura = scanner.nextDouble();
                    System.out.println("Teclee el primer lado del triangulo");
                    double lado1 = scanner.nextDouble();
                    System.out.println("Teclee el segundo lado del triangulo");
                    double lado2 = scanner.nextDouble();
                    System.out.println("Teclee el tercer lado del triangulo");
                    double lado3 = scanner.nextDouble();
                    
                    Triangulo triangulo = new Triangulo(base, altura, lado1, lado2, lado3);
                    System.out.println("El area del triangulo es: " +triangulo.area());
                    System.out.println("El perimetro del triangulo es: " +triangulo.perimetro());
                }
                    
                case 2 -> {
                    System.out.println("Teclee el radio del circulo");
                    double radio = scanner.nextDouble();
                    
                    Circulo circulo = new Circulo(radio);
                    System.out.println("El area del circulo es: " +circulo.area());
                    System.out.println("El perimetro del circulo es: " +circulo.perimetro());
                }
                    
                case 3 -> {
                    System.out.println("Teclee la base del rectangulo");
                    double base = scanner.nextDouble();
                    System.out.println("Teclee la altura del rectangulo");
                    double altura = scanner.nextDouble();
                    Rectangulo rectangulo = new Rectangulo(base,altura);
                    System.out.println("El area del rectangulo es: " +rectangulo.area());
                    System.out.println("El perimetro del rectangulo es: " +rectangulo.perimetro());
                }
                    
                case 4 -> {
                    System.out.println("Teclee el valor de un lado del hexagono");
                    double lado = scanner.nextDouble();
                    System.out.println("Teclee el valor de la apotema");
                    double apotema = scanner.nextDouble();
                    
                    Hexagono hexagono = new Hexagono(lado, apotema);
                    System.out.println("El area del hexagono es:" +hexagono.area());
                    System.out.println("El perimetro del hexagono es: " +hexagono.perimetro());
                }
                    
                case 0 -> System.out.println("Gracias por usar el programa. Hasta luego <3");
                    
                default -> System.out.println("ESA OPCION NO ES VALIDA. Intente con otra");  
            }
        } while(opc != 0);
    }
}

